esp/esp/web/util/latex.py

```diff
@@ -107,18 +109,24 @@ def gen_latex(texcode, type='pdf', remove_files=False, stdout=PIPE, stderr=STDOU
     :rtype:
         HttpResponse
     """
+    files = [stdout, stderr]
```

Comment:
I think this is more complicated than necessary.  You can just pass the same file handle to both inputs if `stdout` and `stderr` are both `devnull_sentinel`, and only open it once.  This avoids the whole `nested()` thing, which I found a bit confusing.